src/Raven.Server/NotificationCenter/TombstoneNotifications.cs

```diff
@@ -0,0 +1,100 @@
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Logging;
+
+namespace Raven.Server.NotificationCenter
+{
+    public class TombstoneNotifications
+    {
+        private readonly NotificationCenter _notificationCenter;
+        private readonly NotificationsStorage _notificationsStorage;
+        private readonly string _database;
+        private readonly Logger _logger;
+        private const string _title = "Blocking of tombstones deletion";
+        private const string _msg = $"We have detected blocking of tombstones deletion. Consider deleting or enabling the following processes:";
+
+        public TombstoneNotifications(NotificationCenter notificationCenter, NotificationsStorage notificationsStorage, string database)
+        {
+            _notificationCenter = notificationCenter;
+            _notificationsStorage = notificationsStorage;
+            _database = database;
+            _logger = LoggingSource.Instance.GetLogger(database, GetType().FullName);
+        }
+
+        public void Add(Dictionary<(string, string), long> blockingTombstones)
+        {
+            BlockingTombstonesDetails details = new BlockingTombstonesDetails(blockingTombstones);
+            _notificationCenter.Add(AlertRaised.Create(_database, _title, _msg, AlertType.BlockingTombstones,
+                NotificationSeverity.Warning,
+                nameof(AlertType.BlockingTombstones), details: details));
+        }
+
+        public List<BlockingTombstoneDetails> GetNotificationDetails(string id)
+        {
+            var list = new List<BlockingTombstoneDetails>();
+            using (_notificationsStorage.Read(id, out var value))
+            {
+                value.Json.TryGet(nameof(AlertRaised.Details), out BlittableJsonReaderObject details);
+                details.TryGet(nameof(BlockingTombstonesDetails.BlockingTombstones), out BlittableJsonReaderArray blockingTombstonesDetails);
+
+                foreach (BlittableJsonReaderObject detail in blockingTombstonesDetails)
+                {
+                    detail.TryGet(nameof(BlockingTombstoneDetails.Source), out string source);
+                    detail.TryGet(nameof(BlockingTombstoneDetails.Collection), out string collection);
+                    detail.TryGet(nameof(BlockingTombstoneDetails.NumberOfTombstones), out long numOfTombstones);
+                    var blockingTombstoneDetails = new BlockingTombstoneDetails()
+                    {
+                        Source = source,
+                        Collection = collection,
+                        NumberOfTombstones = numOfTombstones
+                    };
+                    list.Add(blockingTombstoneDetails);
+                }
+            }
+
+            return list;
+        }
+
+        internal class BlockingTombstonesDetails : INotificationDetails
+        {
+            public BlockingTombstonesDetails(Dictionary<(string, string), long> blockingTombstones)
```

Comment:
Can we add Named parameters here? (and update the callers)
```suggestion
            public BlockingTombstonesDetails(Dictionary<(string Source, string Collection), long> blockingTombstones)
```

Reply:
done.